### DISCUSSION TITLE
Unable to download mitmproxy CA certificate from http://mitm.it

### DISCUSSION THREAD

**Discussion by dmattera** [mitmproxy/mitmproxy#7014] [Q&A]
#### Problem Description
I am trying to download the mitm certificate as defined from the [documentation](https://docs.mitmproxy.org/stable/concepts-certificates/) on iOS 17 using Safari but when I visit `http://mitm.it` it does not bring me to the certificate download page. Instead I get page that just says" If you are seeing this, traffic is not running through mitmproxy"

<img width="1289" alt="Screen Shot 2024-07-09 at 2 45 05 PM" src="https://github.com/mitmproxy/mitmproxy/assets/13613522/9aa20737-830d-4f74-b42e-8edf45838fd2">

#### Steps to reproduce the behavior:
1. go to `http://mitm.it` to try and install certificate
2. no download links are provided

#### System Information
Mitmproxy: 8.0.0 binary
Python:    3.10.2
OpenSSL:   OpenSSL 1.1.1n  15 Mar 2022
Platform:  macOS-10.15.7-x86_64-64bit

I have visited `http://mitm.it` on a number of different platforms and browsers (Safari on iOS 15 & 17, Firefox and Chrome on macOS) and they all are giving the same page that does not have the download links

**Comment by mhils** [comment]
As the error message says: you haven't configured your device to use a proxy server.

**Reply by axxturelmeatrider** [reply]
Hey a little late to the party but I am having the same issue and I have definitely set up a proxy in my settings.

**Reply by axxturelmeatrider** [reply]
![image](https://github.com/user-attachments/assets/edb1e5e6-3a39-40de-8f68-14df75422ee0)
Here are my proxy settings. I don't know why this won't work, please lmk.

**Comment by brandon-fryslie** [comment]
May I ask why it matters?  Is there some sort of technical reason I can't download the cert without my traffic going through the proxy?  If not, any reason this in principle couldn't be a big red banner on the standard cert download page?

I understand my traffic obviously won't be in the proxy and maybe this is to make it really clear that is the case.  However I would still find it convenient to be able to download the cert regardless.  Since I use Wireguard the situation is a bit different and I want to be able to download it from any device.

**Reply by shellvon** [reply]
The mitmproxy certificate must be downloaded through its proxy primarily because:

1.  **Dynamic Generation & Uniqueness:** mitmproxy generates a unique, on-the-fly SSL/TLS certificate for each intercepted HTTPS website, signed by its own unique Certificate Authority (CA) created upon first run.
2.  **Establishing Trust:** Browsers don't inherently trust mitmproxy's CA. Downloading it through the proxy (via `mitm.it` when traffic is proxied) ensures you get the correct CA certificate from the currently running instance to install as trusted, allowing interception without browser warnings.

Offering a direct download wouldn't guarantee you get the right certificate for your current mitmproxy instance and wouldn't guide you through the necessary trust installation process.

More details you can see from docs:
https://docs.mitmproxy.org/stable/concepts-certificates/


The documentation explicitly states that for security reasons (to prevent others from decrypting your HTTPS traffic), a certificate needs to be generated specifically for the current mitmproxy instance. This is the first point. Downloading through the current mitmproxy proxy ensures the certificate's uniqueness.

The documentation also mentions the [manual installation method](https://docs.mitmproxy.org/stable/concepts-certificates/#installing-the-mitmproxy-ca-certificate-manually), providing information about the default generated proxy file format and location.

Hopefully, my answer is helpful to you!
